vc_zoom/indico_vc_zoom/util.py

```diff
@@ -0,0 +1,102 @@
+# This file is part of the Indico plugins.
+# Copyright (C) 2020 CERN and ENEA
+#
+# The Indico plugins are free software; you can redistribute
+# them and/or modify them under the terms of the MIT License;
+# see the LICENSE file for more details.
+
+from __future__ import unicode_literals
+
+import random
+import string
+
+from requests.exceptions import HTTPError
+
+from indico.core.db import db
+from indico.modules.users.models.emails import UserEmail
+from indico.modules.users.models.users import User
+from indico.modules.vc.exceptions import VCRoomError, VCRoomNotFoundError
+from indico.util.date_time import now_utc
+
+from indico_vc_zoom import _
+from indico_vc_zoom.api import ZoomIndicoClient
+
+
+def find_enterprise_email(user):
+    """Find a user's first e-mail address which can be used by the Zoom API.
+
+    :param user: the `User` in question
+    :return: the e-mail address if it exists, otherwise `None`
+    """
+    from indico_vc_zoom.plugin import ZoomPlugin
+    domains = [auth.strip() for auth in ZoomPlugin.settings.get('email_domains').split(',')]
+    # get all matching e-mails, primary first
+    result = UserEmail.query.filter(
```

Comment:
I think you could use `db.session.query(UserEmail.email).***.scalar()` to get the email directly without the separate `result.email if result` step (but I haven't tried it). Nothing wrong with leaving it as it is though!

Reply:
Doesn't `.scalar(...)` blow up if there are no matches?

Reply:
Never mind. Just read the docs.

Reply:
Yeah, works fine, thanks.

Reply:
Ah, actually, it blows up if one has > 1 e-mail which matches.

![image](https://user-images.githubusercontent.com/2699/100237471-4b67e300-2f2f-11eb-936e-9cf1ae608c16.png)

Reply:
ah damn :/ `.limit(1)` might work though

Reply:
Yeah, but at this point I think the original version was easier to understand...